### DISCUSSION TITLE
Problems with routing on different clients

### DISCUSSION THREAD

**Discussion by FB9pq** [strongswan/strongswan#1768] [Q&A]
Hy everyone! 
Have two test clients with same result: clients don't have a route to get to the local network behind a vpn server.
Trying to replicate this example:
https://www.strongswan.org/testing/testresults/ikev2/ip-pool/

"ip route list table 220" shows empty output.

Can ping local ip adresses behind a vpn server only after manually adding a route.

---------------------------------------
swanctl.conf:
-----------------------------------------
connections {
            home {
                remote_addrs = 192.168.100.10
                vips = 0.0.0.0

                local {
                    auth = pubkey
                    certs = carolCert.pem
                    id = carol.strongswan.org
                }
                remote {
                    auth = pubkey
                    id = moon.strongswan.org
                }
                children {
                    home {
                        remote_ts  = 192.168.20.0/24
                        updown = /usr/libexec/ipsec/_updown iptables
                        start_action = start
                    }
                }
            }
        }
----------------------------------------------

strongswan.conf:
--------------------------------------------------
-------------------------------------------------
swanctl {
        load = pem pkcs1 x509 revocation constraints pubkey openssl random
}
charon-systemd {
load = random drbg nonce aes des md5 sha1 sha2 sha3 chapoly pem pkcs1 pkcs8 curve25519 gmp newhope mgf1 ntru x509 curl revocation hmac kdf xcbc ctr ccm gcm vici kernel-netlink socket-default updown
}
-----------------------------------------------------------

**Comment by Kentzo** [comment]
What is in the logs?

**Reply by FB9pq** [reply]
Sure.
[Centos log](https://pastes.io/2tpb2gvmgw)

[Opensuse log](https://pastes.io/urbreqw1xx)

**Reply by Kentzo** [reply]
Hmm, I don’t see policy exchange in the logs.

What xfrm policies were installed on your clients?

**Reply by FB9pq** [reply]
Do you mean kernel-netlink that is mentioned in client's strongswan.conf?
That is all that comes to mind.

**Reply by Kentzo** [reply]
It doesn't looks like you use [route-based](https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html) configuration via VTI / XFRM**i**. Thus it's policies rather than routes that direct traffic via IPsec. Use `ip xfrm state show` and `ip xfrm policy show` after the connection is established.

**Reply by FB9pq** [reply]
Thank you for assistance!
ip xfrm state show - prints nothing

ip xfrm policy show:
src 0.0.0.0/0 dst 0.0.0.0/0 
	socket in priority 0 ptype main 
src 0.0.0.0/0 dst 0.0.0.0/0 
	socket out priority 0 ptype main 
src 0.0.0.0/0 dst 0.0.0.0/0 
	socket in priority 0 ptype main 
src 0.0.0.0/0 dst 0.0.0.0/0 
	socket out priority 0 ptype main 
src ::/0 dst ::/0 
	socket in priority 0 ptype main 
src ::/0 dst ::/0 
	socket out priority 0 ptype main 
src ::/0 dst ::/0 
	socket in priority 0 ptype main 
src ::/0 dst ::/0 
	socket out priority 0 ptype main

**Comment by tobiasbrunner** [comment]
It looks like you are initiating only the IKE_SA. If you are using `swanctl --initiate`, make sure you pass `--child home` to initiate the CHILD_SA.

**Reply by FB9pq** [reply]
You are correct, I didn't know that there is a need for "--child home".
Also I was too lazy to setup firewall for incoming ipsec connections. Decided to disable it completely on the server's side while testing. Soon after the updown script could not apply ipsec rules because of that .
Kentzo have seen it through right from the start by mentioning ipsec policies that direct traffic.
Thank you guys for help!
Strongswan documentation is written by those who know hot to setup strongswan for those who know how to setup strongswan.
